docs/samples/interaction.md

```diff
@@ -0,0 +1,184 @@
+# Interaction
+
+```js chart-editor
+// <block:setup:3>
+const AXIS = ['xy', 'x', 'y'];
+const MODE = ['nearest', 'point', 'x', 'y'];
+
+let axisIndex = 0;
+let modeIndex = 0;
+
+Utils.srand(8);
+
+const data = {
+  labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
+  datasets: [{
+    type: 'line',
+    label: 'Dataset 1',
+    borderColor: 'rgb(54, 162, 235)',
+    borderWidth: 2,
+    fill: false,
+    data: Utils.numbers({count: 7, min: 0, max: 100}),
+  }]
+};
+// </block:setup>
+
+// <block:printer:4>
+const printer = {
+  id: 'printer',
+  afterDraw(chart) {
+    const options = chart.options.plugins.annotation.interaction;
+    const mode = options.mode || 'nearest';
+    const axis = options.axis || 'xy';
+    const intersect = !!options.intersect;
+    const ctx = chart.ctx;
+    ctx.save();
+    ctx.font = '14px monospace';
+    ctx.fillStyle = 'rgba(0, 0, 0, 0.5)';
+    ctx.textAlign = 'left';
+    ctx.textBaseline = 'top';
+    ctx.fillText('Mode: ' + mode + ', axis: ' + axis + ', intersect: ' + intersect, chart.chartArea.left, chart.canvas.offsetHeight - 16);
+    ctx.restore();
+  }
+};
+// </block:printer>
+
+// <block:annotation1:1>
+const annotation1 = {
+  type: 'box',
+  backgroundColor: 'rgba(255, 245, 157, 0.2)',
+  borderColor: 'rgb(255, 245, 157)',
+  borderWidth: 2,
+  enter: function({element}) {
+    console.log(element.options.label.content + ' entered');
+  },
+  click: function({element}) {
+    console.log(element.options.label.content + ' clicked');
+  },
+  leave: function({element}) {
+    console.log(element.options.label.content + ' left');
+  },
+  label: {
+    display: true,
+    content: 'Outer box annotation',
+    position: {
+      y: 'start'
+    },
+    font: {
+      size: 14
+    }
+  },
+  xMax: 'June',
+  xMin: 'February',
+  xScaleID: 'x',
+  yMax: 90,
+  yMin: 10,
+  yScaleID: 'y'
+};
+// </block:annotation1>
+
+// <block:annotation2:2>
+const annotation2 = {
+  type: 'box',
+  backgroundColor: 'rgba(165, 214, 167, 0.2)',
+  borderColor: 'rgb(165, 214, 167)',
+  borderWidth: 2,
+  enter: function({element}) {
+    console.log(element.options.label.content + ' entered');
+  },
+  click: function({element}) {
+    console.log(element.options.label.content + ' clicked');
+  },
+  leave: function({element}) {
+    console.log(element.options.label.content + ' left');
+  },
+  label: {
+    display: true,
+    content: 'Inner box annotation',
+    position: {
+      y: 'start'
+    },
+    font: {
+      size: 14
+    }
+  },
+  xMax: 'May',
+  xMin: 'March',
+  xScaleID: 'x',
+  yMax: 75,
+  yMin: 25,
+  yScaleID: 'y'
+};
+// </block:annotation2>
+
+/* <block:config:0> */
+const config = {
+  type: 'line',
+  plugins: [printer],
```

Comment:
Maby instead of a custom plugin to put the info on the chart you can use a scriptable function for the title. Don't know what is more clear so if this is looks better its also fine

Reply:
good hint! Let me try

Reply:
Plugin has been removed and it uses the title as suggested (really better, thanks a lot @LeeLenaleee).